plugins/modules/systemd_facts.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2025, Marco Noce <nce.marco@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: systemd_facts
+short_description: Gather C(systemd) unit facts.
+description:
+  - This module gathers facts about systemd units (services, targets, sockets).
+  - It runs C(systemctl list-units) (or processes selected units) and collects properties
+    for each unit using C(systemctl show).
+  - Even if a unit has a loadstate of "not-found" or "masked", it is added to the fact,
+    but only with the minimal properties ( name, loadstate, activestate, substate).
+  - When O(select) is used, the module first checks if the unit exists. If not, the unit
+    is recorded with minimal information.
+  - The gathered information is added to ansible facts under the key systemd_units.
+version_added: "10.4.0"
+options:
+  select:
+    description:
+      - When set to V(true), only the units specified in O(unitname) are processed.
+    type: bool
+    default: false
+  unitname:
+    description:
+      - List of unit names to process when O(select) is V(true).
+    type: list
+    elements: str
+    default: []
+  properties:
+    description:
+      - Additional properties to retrieve (appended to the default ones).
+    type: list
+    elements: str
+    default: []
+author:
+  - Marco Noce (@NomakCooper)
+'''
+
+EXAMPLES = r'''
+# Gather facts for all systemd services, targets, and sockets
+- name: Gather all systemd unit facts
+  community.general.systemd_facts:
+
+# Gather facts for selected units with extra properties.
+- name: Gather facts for selected unit(s)
+  community.general.systemd_facts:
+    select: true
+    unitname:
+      - sshd.service
+      - sshd-keygen.target
+      - systemd-journald.socket
+    properties:
+      - Description
+'''
+
+RETURN = r'''
+ansible_facts:
+  description: Dictionary of systemd unit facts keyed by unit name.
+  returned: always
+  type: complex
+  contains:
+    systemd_units:
+      description: Properties of systemd units
+      returned: always
+      type: list
+      elements: dict
+      contains:
+        name:
+          description: Unit name.
+          returned: always
+          type: str
+          sample: sshd.service
+        loadstate:
+          description:
+          - The state of the unit's configuration load
```

Comment:
```suggestion
          - The state of the unit's configuration load.
```
(There are some more of these.)